util/gitutil/detectgit.go

```diff
@@ -284,9 +284,13 @@ func gitRelDir(basePath string, path string) (string, bool, error) {
 	if err != nil {
 		return "", false, errors.Wrapf(err, "stat for %s", basePath)
 	}
-	// `pathParts` here has lose the root filepath.Separator since built by strings.Split beforehand.
-	// so putting heading separator is required to make it absolute again.
-	b, err := os.Stat(string(filepath.Separator) + filepath.Join(pathParts[:len(basePathParts)]...))
+	rootPath := ""
+	if filepath.VolumeName(basePath) == "" {
+		// `pathParts` here has lose the root filepath.Separator since built by strings.Split beforehand.
+		// so putting heading separator is required to make it absolute again.
```

Comment:
This is only valid reasoning in case the basePath does not have a volume name (i.e.: C:), luckily filepath provides an easy way to detect when the path does not have a volume name.

Reply:
Seems reasonable to me.